Update rust dependencies

jpculp opened the pull request:
**Description of changes:**

Ran `cargo update` for `sources` and `tools`, avoided some updates that pull unnecessary extra versions, and adjusted the deny TOMLs as necessary. Also included updates for **AWS SDK for Rust**, `coldsnap`, and `tough`.

**Testing done:**

- [x] Build Bottlerocket
- [x] Publish AMI
- [x] Create SSM Parameter


**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

jpculp commented:
I'll split this up into multiple commits after `tough` and `coldsnap` are released.

jpculp commented on tools/deny.toml:
```diff
@@ -65,25 +65,30 @@ multiple-versions = "deny"
     { name = "wasi", version = "=0.10.2" },
+    # aws-sdk-rust is using an old version of fastrand
+    { name = "fastrand", version = "=1.9" },
+    # aws-sdk-rust is using an old version of rustls, hyper-rustls, and tokio-rustls
+    { name = "rustls", version = "=0.20" },
+    { name = "hyper-rustls", version = "=0.23" },
+    { name = "tokio-rustls", version = "=0.23" },
+    # kube-client uses an old version of redox_syscall
+    { name = "redox_syscall", version = "=0.2" },
```
This looks like a lot, but it's because we're no longer skipping the entire `testsys-model` and `bottlerocket-types` trees.

webern reviewed (approved):
Looks good. Actually looked at it.

jpculp committed:
sources: bump nix from 0.24 to 0.26

Also bumps gptman from 1.0.0 to 1.0.1.

jpculp committed:
sources: update aws-sdk-rust and tough

jpculp committed:
tools: update aws-sdk-rust, coldsnap, and tough

jpculp committed:
sources: update rust dependencies

jpculp committed:
tools: update rust dependencies

jpculp committed:
makefile: bump tuftool to 0.10.0

jpculp commented:
- Switched to newly released `tough` and `coldsnap` crates.
- Broke PR up into multiple commits.
- Updated a couple more minor dependencies.
- Bumped **PUBLISH_TUFTOOL_VERSION** in `Makefile.toml`.

jpculp commented:
Rebased PR. Ready for review.

stmcginnis reviewed (approved):
Built variant and able to deploy. Everything looks good.
